src/main/java/io/druid/data/input/MapBasedRow.java

```diff
@@ -164,11 +151,8 @@ public boolean equals(Object o)
     if (!event.equals(that.event)) {
       return false;
     }
-    if (!timestamp.equals(that.timestamp)) {
-      return false;
-    }
+    return timestamp.equals(that.timestamp);
```

Comment:
Unless there is a specific reason to optimize this code, this is probably unnecessary. We typically use IntelliJ to generate to equals code, which means this change will get discarded next time someone needs to re-generate the equals method.